### PR TITLE
[MESSENGER] Async emails

### PR DESCRIPTION
**Pull request by Guikingone**
Small addition in order to explain how the messenger can be used in order to send emails asynchronously using a message broker.

Need to be reviewed for syntax and sentences.

Thanks to @sroze for the help 😄

### CODE REVIEW

**Comment by OskarStark** [comment]
Hmm, not sure its worth to have such a big block of text, just for deactivating the spool...
It could be done using a note block

**Comment by Guikingone** [comment]
@OskarStark Yep, I'm gonna correct this asap.

**Review by sroze — APPROVED** [review]

File: messenger.rst [hunk 1]
```diff
@@ -371,6 +371,45 @@ while still having them passed to their respective handler:
             ),
         ));
 
+.. tip::
+
+    If you need to send asynchronous emails, it could be usefull to deactivate
+    the spool as it's only called during the ``kernel.terminate`` event (which
+    is not dispatched during command call).
+
+    The first thing to change is the SwiftMailer configuration:
```
Comment (OskarStark) [inline messenger.rst]: You talk about first but no second one 🤔

**Comment by Guikingone** [comment]
@OskarStark Fixed with the second block about the message 😃

**Comment by Guikingone** [comment]
Ok, a small update has been made about the `first` usage as signaled by @OskarStark.

File: messenger.rst [hunk 2]
```diff
@@ -371,6 +371,52 @@ while still having them passed to their respective handler:
...
+            <container xmlns="http://symfony.com/schema/dic/services"
+                xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                xmlns:swiftmailer="http://symfony.com/schema/dic/swiftmailer"
+                xsi:schemaLocation="http://symfony.com/schema/dic/services
+                    http://symfony.com/schema/dic/services/services-1.0.xsd
+                    http://symfony.com/schema/dic/swiftmailer
+                    http://symfony.com/schema/dic/swiftmailer/swiftmailer-1.0.xsd">
+
+                <swiftmailer:config>
+                    <!-- Other keys expect the spool one -->
```
Comment (Lyrkan) [inline messenger.rst]:
```suggestion
                    <!-- Other keys except the spool one --> 
```

File: messenger.rst [hunk 3]
```diff
@@ -371,6 +371,52 @@ while still having them passed to their respective handler:
...
+                <swiftmailer:config>
+                    <!-- Other keys expect the spool one --> 
+                </swiftmailer:config>
+            </container>
+
+        .. code-block:: php
+
+            // config/packages/messenger.php
+            $container->loadFromExtension('swiftmailer', array(
+                  // ... Other keys expect the spool one.
```
Comment (Lyrkan) [inline messenger.rst]:
```suggestion
                  // ... Other keys except the spool one.
```

File: messenger.rst [hunk 4]
```diff
@@ -371,6 +371,52 @@ while still having them passed to their respective handler:
             ),
         ));
 
+.. tip::
+
+    If you need to send asynchronous emails, it could be usefull to deactivate
```
Comment (Lyrkan) [inline messenger.rst]:
```suggestion
    If you need to send asynchronous emails, it could be useful to deactivate
```

File: messenger.rst [hunk 5]
```diff
@@ -371,6 +371,52 @@ while still having them passed to their respective handler:
...
+            </container>
+
+        .. code-block:: php
+
+            // config/packages/messenger.php
+            $container->loadFromExtension('swiftmailer', array(
+                  // ... Other keys expect the spool one.
+            ));
+            
+        Then time to send your message which contain the email content:
```
Comment (Lyrkan) [inline messenger.rst]:
```suggestion
        Then time to send your message which contains the email content:
```

**Review by Lyrkan — CHANGES_REQUESTED** [review]

**Comment by Guikingone** [comment]
@javiereguiluz It seems that SfCloud fails to deploy the actual revision, normal situation?

**Comment by javiereguiluz** [comment]
@Guikingone it's not normal ... so let's ping @tucksaun to see if he can take a look. Thanks!

**Comment by tucksaun** [comment]
@Guikingone this PR is quite old so this branch does not contain the required files for SymfonyCloud. Try to rebase onto the latest 4.1 commit and it should work.

**Comment by javiereguiluz** [comment]
I'm closing this old pull request because we recently added something like this for the new mailer component. See https://symfony.com/doc/current/mailer.html#sending-messages-async  Thanks.

**Pull request closed without merging**